mail/views.py

```diff
@@ -237,3 +238,37 @@ def post(self, request, *args, **kwargs):
             status=mailgun_response.status_code,
             data=generate_mailgun_response_json(mailgun_response)
         )
+
+
+class EmailBouncedView(APIView):
+    """
+    View class that handles HTTP requests to course team mail API
+    """
+    permission_classes = (permissions.AllowAny, )
+
+    @classmethod
+    def is_course_team_email(cls, recipient):
+        """checks if recipient is valid course team email"""
```

Comment:
Instead of comparing the email addresses here can we mark the original email being sent as an email which is not allowed to bounce? We can send this data in the `custom variables` part of the message, and receive it as a header variable in this endpoint.

Reply:
can you explain this. @noisecapella

Reply:
All emails are sent via this function: https://github.com/mitodl/micromasters/blob/master/mail/api.py#L91
You should add an argument `error_on_bounce` which defaults to `True` but which is set to `False` for `send_course_team_email`.

You can attach data to a message like this: https://documentation.mailgun.com/en/latest/user_manual.html#attaching-data-to-messages

You should pass the `error_on_bounce` value in the data here. You can look in the header of this webhook for this information and use it to determine if you should log the bounce or ignore it.